nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/broadsides.dm

```diff
@@ -166,3 +265,147 @@
 
 /obj/effect/particle_effect/muzzleflash/Destroy()
 	return ..()
+
+//Newer Broadsides features start here
+
+/obj/machinery/ship_weapon/broadside/MouseDrop_T(atom/movable/A, mob/user)
+	if(stovepipe)
+		to_chat(user, "<span class='warning'>The [src] is completely locked up, you have to <i>pry</i> out the stovepiped shell!</span>")
+		return FALSE
+	if(ismoth(A) || istype(A, /mob/living/simple_animal/mothroach && A != user))
+		mothclean(A, user)
+		return TRUE
+	if(prob(soot)) //likelihood of stovepipe is a lot higher if you crateload the gun
+		stovepipe = TRUE
+		add_overlay("[dir]_broadside_stovepipe")
+		flick("[initial(icon_state)]_chambering",src)
+		playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+		to_chat(user, "<span class='warning'>The [src] groans horrendously, a shell has stovepiped!</span>")
+		return FALSE
+
+	..()
+	if(!isliving(user))
+		return FALSE
+	if(istype(A, /obj/structure/closet))
+		if(!locate(/obj/item/ship_weapon/ammunition/broadside_shell) in A.contents)
+			to_chat(user, "<span class='warning'>There's nothing in [A] that can be loaded into [src]...</span>")
+			return FALSE
+		if(length(ammo) >= max_ammo)
+			return FALSE
+		to_chat(user, "<span class='notice'>You start to load [src] with the contents of [A]...</span>")
+		if(do_after(user, 8 SECONDS , target = src))
+			for(var/obj/item/ship_weapon/ammunition/broadside_shell/G in A)
+				if(length(ammo) < max_ammo)
+					G.forceMove(src)
+					ammo += G
+					chambered = G
+			if(feeding_sound)
+				playsound(src, feeding_sound, 100, 1)
+			if(load_sound)
+				playsound(src, load_sound, 100, 1)
+			flick("[initial(icon_state)]_chambering",src)
+			cut_overlay("[dir]_chambered_[get_ammo()]")
+			add_overlay("[dir]_chambered_[get_ammo()]")
+			state = STATE_CHAMBERED
+			update()
+
+
+/obj/machinery/ship_weapon/broadside/proc/mothclean(mob/moth, mob/user) //This is an easter egg, let's you clean the gun with moth/roach
+	if(!panel_open)
+		to_chat(user, "<span class='notice'>You can't reach the barrels unless you <i>unscrew</i> the maintenance panel.</span>")
+		return FALSE
+	if(soot == 0)
+		to_chat(user, "<span class='notice'>The [src] is clean as can be, no need for further swabbing.</span>")
+		return FALSE
+	if(do_after(user, 5 SECONDS, target = src))
```

Comment:
Missing safety checks accounting for e.g. the other individual moving away given target is the weapon.